plugins/file-open-save-new/core/src/test/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProviderTest.java

```diff
@@ -0,0 +1,215 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2024 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.plugins.fileopensave.providers.vfs;
+
+import junit.framework.TestCase;
+import org.junit.Test;
+import org.pentaho.di.connections.ConnectionDetails;
+import org.pentaho.di.connections.vfs.VFSConnectionDetails;
+import org.pentaho.di.plugins.fileopensave.api.providers.File;
+import org.pentaho.di.plugins.fileopensave.providers.vfs.model.VFSFile;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class VFSFileProviderTest extends TestCase {
+
+  public void testIsConnectionRoot() {
+
+    VFSFileProvider testInstance = createTestInstance();
```

Comment:
Should include tests with connection names with special characters.

Reply:
Addressed see commit https://github.com/pentaho/pentaho-kettle/commit/3e8bcc9f08d5634a303f5bdcbaf73f80d1811e46